### PR TITLE
Feature/create page save function 2

### LINKED ISSUE #22: Create page / save function / 2
## Value Proposition
As a tutor, I want the tutorial data I enter to be saved so that it can be used in the future.

## Description
capstone-userstory-08

## Acceptance Criteria
- [x] when the `finish creating` button is clicked the entered data is saved to a database and the user will be redirected to the `library` page

## Tasks
- [x] create branch
- [x] set up and implement MongoDB
- [x] show database content on `library` page 
- [x] write new tutorial data to database
- [x] implement redirection to `library` page

### PR DESCRIPTION
**Pull request by kai-franke**
In this PR I have implemented a MongoDB. I can now read from the database and save a new tutorial.

### CODE REVIEW

**Comment by JannikUekoetter** [comment]
Dear @kai-franke, you have really succeeded in creating a masterpiece. This is a formidable piece of work. The functionality is absolutely there, and you can be very proud of that.

**Review by JannikUekoetter — APPROVED** [review]

**Comment by kai-franke** [comment]
> Dear @kai-franke, you have really succeeded in creating a masterpiece. This is a formidable piece of work. The functionality is absolutely there, and you can be very proud of that.

Thank you for your warm words, @JannikUekoetter !

File: components/CreateForm.js [hunk 1]
```diff
@@ -54,17 +74,26 @@ function CreateForm() {
...
     }
+    const stepsWithModifiedUrls = inputSteps.map((inputStep) => {
+      return {
+        ...inputStep,
+        img: addProxyToImgUrl(inputStep.img),
+      };
+    });
+
     const newTutorial = {
       id: nanoid(),
```
Comment (ThomasF85) [inline components/CreateForm.js]: Why create an id here when you send it to mongoDB as the next step (which will automatically set an id for the object)?

File: models/Tutorial.js [hunk 2]
```diff
@@ -0,0 +1,16 @@
+import mongoose from "mongoose";
+
+const { Schema } = mongoose;
+
+const tutorialSchema = new Schema({
+  id: { type: String, required: true },
```
Comment (ThomasF85) [inline models/Tutorial.js]: Why set an id here? You will end up with two id fields in the database, because an _id field will automatically be created.

**Review by ThomasF85 — APPROVED** [review]
Looks very good! But please look into why you are using another id field in your database.

File: components/CreateForm.js (hunk 1 above)
Comment (kai-franke) [reply components/CreateForm.js]: Good point! It was left over from previous steps.

File: models/Tutorial.js (hunk 2 above)
Comment (kai-franke) [reply models/Tutorial.js]: It's removed now.

**Pull request merged**